content/blog/2024/jupyterhub-binderhub-gesis/index.md

```diff
@@ -0,0 +1,126 @@
+
+## Explored pathways that were then discarded
+
+List of things that were tried and then decided as not good pathways:
+
+- [repo2docker-service](https://github.com/consideRatio/repo2docker-service), a separate JupyterHub service that could *only* build images. As we worked on it, we realized that it was replicating a lot of features that BinderHub already has, so we pivoted to working on BinderHub directly instead.
+- Building off of [tljh-repo2docker](https://github.com/plasmabio/tljh-repo2docker). While this already had a nice UI, it would be hard to port it to run on a distributed kubernetes environment without it becoming a 'hard fork'.
+
+While these did slow down the implementation of the project, it has allowed us to be very confident that the methods we have chosen are long term sustainable.
+
+## Want to try this out?
+
+We have a demo of this running at [imagebuilding-demo.2i2c.cloud](https://imagebuilding-demo.2i2c.cloud), but unfortunately as we are still fine-tuning `cryptnono` config, at this moment it is not open to the public. Please [hit me up](mailto:yuvipanda@2i2c.org) with your GitHub account if you want access, and promise to not be a cryptominer and you shall be granted access.
+
+Want to set this up on your own JupyterHub? There is some [work in progress](https://github.com/2i2c-org/2i2c-org.github.io/pull/205) documentation and more is being worked on. Drop a line in the linked pull request and we'll be happy to help. The eventual goal is for *anyone* to be able to simply follow documentation and set this up for themselves.
+
+## Future work
+
+This is not complete of course, and there is a lot of future work to be done.
+
+1. mybinder.org also helps you distribute your *content*, not just the environment for your code to run in. Since JupyterHub usually comes with a persistent home directory for the user, [nbgitpuller](https://github.com/jupyterhub/nbgitpuller/) is commonly used for this purpose instead. We should explore ways to integrate nbgitpuller (and other ways to distribute content) in the future.
+2. More thorough documentation for how you can recreate what is in the demo for yourself in your own JupyterHub installation.
```

Comment:
@yuvipanda let me know where I can help with this 😄

Reply:
Recalling an excellent conversation with @yuvipanda at his place in Oakland, I wish to imagine that `nbgitpuller link` and `binder link` are really the same thing. These links incorporate a `target hub url` and a `content source` with an environment specification and a landing location. When clicked, users are brought to the target hub url with the specified environment, inside the content, and at the landing location. Whether the experience is "binder like" or "JupyterHub like" is determined by the whether the target hub provides ephemeral or persistent storage. 

It may not be appropriate to highlight in this post but I feel that Yuvi's view that BinderHub = JupyterHub (with a storage boolean flipped to _ephemeral_ instead of _persistent_) needs to be conveyed more widely.

Reply:
Yes, I agree it would be useful to convey that more widely. Not sure what's the appropriate medium though - not this blog post I think. I'm glad that conversation was useful, @colliand!